acvm/src/lib.rs

```diff
@@ -117,8 +137,14 @@ pub trait PartialWitnessGenerator {
     fn solve_directives(
         initial_witness: &mut BTreeMap<Witness, FieldElement>,
         directive: &Directive,
-    ) -> Result<(), OpcodeResolutionError> {
-        pwg::directives::solve_directives(initial_witness, directive)
+    ) -> Result<GateResolution, OpcodeResolutionError> {
+        match pwg::directives::solve_directives(initial_witness, directive) {
+            Ok(_) => Ok(GateResolution::Solved),
+            Err(OpcodeResolutionError::OpcodeNotSolvable(unsolved)) => {
+                Ok(GateResolution::Stalled(unsolved))
+            }
+            Err(err) => Err(err),
+        }
```

Comment:
This smells like we're leaking a concept which is only relevant to `blocks.solve()` into the other types of opcode.

Reply:
This the part where we do the distinction between real errors and GateResolution. `Stalled` and `Solved` are relevant to all opcodes.